profiler/src/main/java/com/splunk/opentelemetry/profiler/JfrRecorder.java

```diff
@@ -16,24 +16,96 @@
 
 package com.splunk.opentelemetry.profiler;
 
+import com.google.common.annotations.VisibleForTesting;
+import java.io.IOException;
+import java.nio.file.Path;
+import java.time.Duration;
+import java.time.Instant;
+import java.util.Map;
+import jdk.jfr.Recording;
+import jdk.jfr.RecordingState;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 /** Responsible for starting a single JFR recording. */
 public class JfrRecorder {
   private static final Logger logger = LoggerFactory.getLogger(JfrRecorder.class.getName());
+  static final String RECORDING_NAME = "otel_agent_jfr_profiler";
+  private final JfrSettingsReader settingsReader;
+  private final Duration maxAgeDuration;
+  private final JFR jfr;
+  private Recording recording;
+
+  JfrRecorder(Builder builder) {
+    this.settingsReader = builder.settingsReader;
+    this.maxAgeDuration = builder.maxAgeDuration;
+    this.jfr = builder.jfr;
+  }
 
   public void start() {
     logger.debug("Profiler is starting a JFR recording");
+    recording = newRecording();
+    Map<String, String> settings = settingsReader.read();
+    recording.setSettings(settings);
+    recording.setToDisk(false);
+    recording.setName(RECORDING_NAME);
+    recording.setDuration(null); // record forever
+    recording.setMaxAge(maxAgeDuration);
+    recording.start();
+  }
+
+  @VisibleForTesting
+  Recording newRecording() {
+    return new Recording();
   }
 
   public void flushSnapshot() {
-    logger.debug("Flushing a snapshot");
+    try {
+      Recording snap = jfr.takeSnapshot();
+      Path path = Path.of(Instant.now().toString() + ".jfr");
+      logger.debug("Flushing a JFR snapshot: " + path);
```

Comment:
Please, don't use string concatenation in logging.

Reply:
I think I missed a boat. 

Why?  I've seen Is templating elsewhere, but didn't imagine that it would be lower cost.  Happy to change it, but want to understand the intention.

Reply:
I think that main reason is that if the logger is set to INFO, you'll still pay the (admittedly small) cost of the string concat even if nothing gets logged.

Reply:
OH!  Derp, yes, that is a very good reason.  🌹

Reply:
Thank you.